lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -461,42 +468,66 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
     end
   end
 
-  defp scan_binding_async(%{scan_binding: nil} = info, _state), do: info
+  defp scan_binding_async(_ref, %{scan_binding: nil} = info, _state), do: info
 
-  defp scan_binding_async(info, state) do
-    # The requests are handled asynchronously and may target different
-    # evaluators, so we include the version that the receiver can check
-    # to avoid race conditions
-    info = update_in(info.scan_binding_version, &(&1 + 1))
-    %{pid: pid, scan_binding: scan_binding, scan_binding_version: version} = info
+  defp scan_binding_async(ref, %{scan_binding_worker: pid} = info, state) when pid != nil do
+    if Process.alive?(pid) do
```

Comment:
I think this Process.alive? check is a smell, but I am not a 100% sure. I assume you are monitoring the smart cell/evaluator process, if you are not, then you should as that is more reliable than `try/catch` (a try/catch won't handle broken links, for example).

In any case, it is best to have everything go through a single code path (pending -> receive a exit/done message) then provide a fast path here and have two execution paths. :)

Reply:
The scanning process is either evaluator or a temporary task, we are waiting for it to finish before doing another scanning. I was monitoring initially and then simplified to this check (if the process died it no longer blocks the scanning), but now realised that if the process dies we may want to do another scanning right away, so monitoring is better (and a single code path sounds better too :)).

Reply:
Great, in case it helps, Process.alive? is almost always a bad idea. :D once this is finished, pls ship it!